addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/payloader/ExtensionPayloader.java

```diff
@@ -0,0 +1,97 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha.payloader;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.zap.extension.custompayloads.ExtensionCustomPayloads;
+import org.zaproxy.zap.extension.custompayloads.PayloadCategory;
+import org.zaproxy.zap.extension.pscanrulesAlpha.JSFunctionPassiveScanner;
+
+public class ExtensionPayloader extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionPayloaderPscanRulesAlphaRelease";
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+    private static ExtensionCustomPayloads ecp;
+    private PayloadCategory jsFuncCategory;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(1);
+        dependencies.add(ExtensionCustomPayloads.class);
+        DEPENDENCIES = Collections.unmodifiableList(dependencies);
+    }
+
+    public ExtensionPayloader() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        ecp =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionCustomPayloads.class);
+        jsFuncCategory =
+                new PayloadCategory(
+                        JSFunctionPassiveScanner.JS_FUNCTION_PAYLOAD_CATEGORY,
+                        JSFunctionPassiveScanner.DEFAULT_FUNCTIONS);
+        ecp.addPayloadCategory(jsFuncCategory);
+        JSFunctionPassiveScanner.setPayloadProvider(() -> jsFuncCategory.getPayloadsIterator());
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        JSFunctionPassiveScanner.setPayloadProvider(null);
+        ecp.removePayloadCategory(jsFuncCategory);
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    @Override
+    public String getAuthor() {
+        return Constant.ZAP_TEAM;
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("pscanrules.payloader.desc");
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("pscanrules.payloader.name");
+    }
+}
```

Comment:
You'll need to modify the pscanrulesAlpha Messages.properties to setup these keys/values.

Reply:
Ah okay gotcha, I had this nagging feeling I was forgetting something, thanks!